### ISSUE TITLE
Crime Monitor not recording factions

### ISSUE THREAD

**Issue by Darkcyde13** [EDCD/EDDI#2160]
## What's Wrong (please be as specific as possible)
When earning bounties, the crime monitor is not recording factions, and thus fills the crime monitor log with an entry for each individual bounty.

### Expected
The Crime Monitor to log all bounties for each faction as a single entry, with the faction name.

### Observed
The Crime monitor is making one entry for each bounty, with no faction.

### Steps to reproduce
1. Kill a wanted ship that has a bounty.
2. Check the Crime Monitor tab in EDDI.
3. Observe an entry for the bounty amount with no faction assigned to it, and no information other than the credit amount.

### Configuration
* Version [Please give full version numbers here, not just "latest" or "current"]
  * EDDI Version: v3.7.3
  * VoiceAttack Version: v1.8.7.12_beta
* Localization [if applicable]
  * EDDI's language: English
  * Voice: Ivona Amy British English

### Investigation Notes
I noticed this a week or two ago, but didn't think much of it at the time, as I knew ED doesn't always provide enough info to log a bounty.  However, after CMDR FlySwat mentioned it on the [forums,](https://forums.frontier.co.uk/threads/eddi-3-3-bring-your-cockpit-to-life.387955/post-9106070) I looked into it further.

Looking over my logs, I can see about 4 seconds after earning a bounty (at least for the first couple I looked at), I get a network error:
`[Warning] Net:GetResponse Failed to obtain response, error code Timeout`
But I'm not sure if that's relevant, it may just be a coincidence.

The other thing I noticed in the crimemonitor,json, is that the faction name is recorded as "Enter Faction Name", and the actual faction is recorded under the "victim" heading.  I would have thought that the victim would have been the name of the NPC I killed, rather than the faction.  Does this mean that the parsing of the Journal entry is not working as expected?

I want to try the previous version of EDDI, to see if that has the same problem, but I won't have time to try that until tomorrow now.

### EDDI Logs
[eddi1.log](https://github.com/EDCD/EDDI/files/6359686/eddi1.log)
[crimemonitor.zip](https://github.com/EDCD/EDDI/files/6359699/crimemonitor.zip)

### Player journals
[Journal.210421181633.01.log](https://github.com/EDCD/EDDI/files/6359715/Journal.210421181633.01.log)

**Comment by ITZAP** [comment]
I play as CMDR FlySwat and experienced these Crime Monitor and Response Timeout issues.
Crime Monitor was working fine in EDDI v3.7.3 until just recently.
Attached please find my EDDI logs and other JSON files for your analysis.
![eddi-crime-monitor](https://user-images.githubusercontent.com/17208120/115798836-8c9aeb00-a41a-11eb-8f14-952ad024fc3f.jpg)
[EDDI-error-report-april-2021.zip](https://github.com/EDCD/EDDI/files/6361851/EDDI-error-report-april-2021.zip)

**Comment by Darkcyde13** [comment]
Interesting.  I wonder... could it be that EDDI is trying to get information about the faction from that server destroyed in a fire recently?  Maybe it's trying to get the factions home system and station to add to the Crime Monitor entry for the bounty, and when it fails, it can't make the entry properly?  Of course, that's just a guess, but it sounds like a plausible explanation.

**Comment by Tkael** [comment]
"Enter Faction Name" provides a clue - that string is only set in two locations - when a user is manually updating a faction entry and when we don't think that we've been able to obtain full faction details from the internet (something that we always do when adding a new faction record).

In this case, the issue stems from a change made to a different (3rd party) server we use to obtain faction details for the faction record (to identify an appropriate system and station for interacting with that faction). Two weeks ago, that server deprecated the API endpoint we were using. That triggered the code to forget the faction name and instead prompt the user for a faction name.
https://github.com/EDCD/EDDI/blob/608712d25adc12c5f8d56c115e8c26fde6dbc128/CrimeMonitor/CrimeMonitor.cs#L935-L943
We've already incorporated the updated API for the next release and I'll modify the code here to preserve the faction name if we already know it, even if we don't have any other details (like EDDBID) from the 3rd party server.
